src/PrestaShopBundle/Controller/Admin/Improve/Modules/ModuleAbstractController.php

```diff
@@ -89,6 +92,42 @@ protected function getToolbarButtons()
             ];
         }
 
-        return $toolbarButtons;
+        return array_merge($toolbarButtons, $this->getExtraToolbarButtons());
+    }
+
+    /**
+     * This method will call the actionAdminModuleExtraToolbarButton hook and allow
+     * modules to add some extra buttons to the modules catalogue toolbar.
+     * All the button's specification keys ('href', 'desc', 'icon', 'help') are mandatory
+     */
+    private function getExtraToolbarButtons(): array
+    {
+        try {
+            $extraToolbarContentFromHooks = (new HookFinder())
+                ->setHookName('actionAdminModuleExtraToolbarButton')
+                ->setParams(['controller' => $this])
+                ->find();
+        } catch (CoreException $exception) {
+            return [];
```

Comment:
it's a bit quiet, we may have to throw a new exception, catch it in the controller and display an error flash message in the view.

Reply:
I don't want modules' additional buttons to avoid displaying Core's regular buttons